### PR TITLE
Improve docs on security http headers

### PR DESCRIPTION
**Pull request by dima-vm**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1957,6 +1957,9 @@ General security recommendations:
 - All the VictoriaMetrics components must run in protected private networks without direct access from untrusted networks such as Internet. The exception is [vmauth](https://docs.victoriametrics.com/vmauth.html) and [vmgateway](https://docs.victoriametrics.com/vmgateway.html).
 - All the requests from untrusted networks to VictoriaMetrics components must go through auth proxy such as vmauth or vmgateway. The proxy must be set up with proper authentication and authorization.
 - Prefer using lists of allowed API endpoints, while disallowing access to other endpoints when configuring auth proxy in front of VictoriaMetrics components.
+- Set reasonable [`Strict-Transport-Security`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security) header value to all the components to mitigate MitM attacks, for example: `max-age=31536000; includeSubDomains`. See `-http.header.hsts` flag.
```
Comment (hagen1778) [inline README.md:1960]: could it be `to public facing` instead?

Comment (hagen1778) [reply README.md:1960]: would be also add a link to MitM attack

File: README.md [hunk 2]
```diff
@@ -1957,6 +1957,9 @@ General security recommendations:
...
+- Set reasonable [`Content-Security-Policy`](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) header value to mitigate XSS attacks. See `-http.header.csp` flag.
```
Comment (hagen1778) [inline README.md:1961]: would be nice to add a link about XSS attack

File: README.md [hunk 3]
```diff
@@ -1957,6 +1957,9 @@ General security recommendations:
...
+- Set reasonable [`X-Frame-Options`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options) header value to mitigate clickjacking attacks, for example `DENY`. See `-http.header.frameOptions` flag.
```
Comment (hagen1778) [inline README.md:1962]: would be nice to add a link about clickjacking attacks

File: lib/httpserver/httpserver.go [hunk 4]
```diff
@@ -52,9 +52,9 @@ var (
 	idleConnTimeout             = flag.Duration("http.idleConnTimeout", time.Minute, "Timeout for incoming idle http connections")
 	connTimeout                 = flag.Duration("http.connTimeout", 2*time.Minute, `Incoming http connections are closed after the configured timeout. This may help to spread the incoming load among a cluster of services behind a load balancer. Please note that the real timeout may be bigger by up to 10% as a protection against the thundering herd problem`)
 
-	headerHSTS         = flag.String("http.header.hsts", "", "Value for 'Strict-Transport-Security' header")
-	headerFrameOptions = flag.String("http.header.frameOptions", "", "Value for 'X-Frame-Options' header")
-	headerCSP          = flag.String("http.header.csp", "", "Value for 'Content-Security-Policy' header")
+	headerHSTS         = flag.String("http.header.hsts", "", "Value for 'Strict-Transport-Security' header, recommended: `max-age=31536000; includeSubDomains`")
+	headerFrameOptions = flag.String("http.header.frameOptions", "SAMEORIGIN", "Value for 'X-Frame-Options' header")
```
Comment (valyala) [inline lib/httpserver/httpserver.go:56]: Let's change the default value for `http.header.frameOption` in a separate pull request, since it may potentially result in bug at client side. So it is better to make this change in a separate pull request, so it could be reverted cleanly if needed.

Comment (valyala) [inline lib/httpserver/httpserver.go:56]:
```suggestion
	headerFrameOptions = flag.String("http.header.frameOptions", "", "Value for 'X-Frame-Options' header")
```

File: README.md (hunk 1 above)
Comment (valyala) [inline README.md:1960]:
```suggestion
- Set reasonable [`Strict-Transport-Security`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security) header value to all the components to mitigate [MitM attacks](https://en.wikipedia.org/wiki/Man-in-the-middle_attack), for example: `max-age=31536000; includeSubDomains`. See `-http.header.hsts` flag.
```

File: README.md (hunk 2 above)
Comment (valyala) [inline README.md:1961]:
```suggestion
- Set reasonable [`Content-Security-Policy`](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) header value to mitigate [XSS attacks](https://en.wikipedia.org/wiki/Cross-site_scripting). See `-http.header.csp` flag.
```

File: README.md (hunk 3 above)
Comment (valyala) [inline README.md:1962]:
```suggestion
- Set reasonable [`X-Frame-Options`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options) header value to mitigate [clickjacking attacks](https://en.wikipedia.org/wiki/Clickjacking), for example `DENY`. See `-http.header.frameOptions` flag.
```

**Review by valyala — APPROVED** [review]

**Review by valyala — APPROVED** [review]

**Comment by valyala** [comment]
@dima-vm , thanks for the pull request!
